src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs

```diff
@@ -322,6 +325,23 @@ internal void ToString(TraceFormat traceFormat, StringBuilder sb)
                             sb.Append(' ');
                             sb.AppendFormat(CultureInfo.InvariantCulture, inFileLineNum, fileName, sf.GetFileLineNumber());
                         }
+                        else if (LocalAppContextSwitches.ILOffsetToStackTrace)
+                        {
+                            try
+                            {
+                                BinaryPrimitives.WriteInt32BigEndian(bytes, mb.MetadataToken);
```

Comment:
Formatting of the hex number should be done as part of the AppendFormat string. Doing it manually like this is both more code and less efficient.